barretenberg/cpp/src/barretenberg/sumcheck/sumcheck_round.hpp

```diff
@@ -104,33 +104,36 @@ template <typename Flavor> class SumcheckProverRound {
     template <typename ProverPolynomialsOrPartiallyEvaluatedMultivariates>
     void extend_edges(ExtendedEdges& extended_edges,
                       ProverPolynomialsOrPartiallyEvaluatedMultivariates& multivariates,
-                      size_t edge_idx,
-                      std::optional<ZKSumcheckData<Flavor>> zk_sumcheck_data = std::nullopt)
+                      size_t edge_idx)
     {
+        for (auto [extended_edge, multivariate] : zip_view(extended_edges.get_all(), multivariates.get_all())) {
+            bb::Univariate<FF, 2> edge({ multivariate[edge_idx], multivariate[edge_idx + 1] });
+            extended_edge = edge.template extend_to<MAX_PARTIAL_RELATION_LENGTH>();
+        }
+    }
 
-        if constexpr (!Flavor::HasZK) {
-            for (auto [extended_edge, multivariate] : zip_view(extended_edges.get_all(), multivariates.get_all())) {
-                bb::Univariate<FF, 2> edge({ multivariate[edge_idx], multivariate[edge_idx + 1] });
-                extended_edge = edge.template extend_to<MAX_PARTIAL_RELATION_LENGTH>();
-            }
-        } else {
-            // extend edges of witness polynomials and add correcting terms
-            for (auto [extended_edge, multivariate, masking_univariate] :
-                 zip_view(extended_edges.get_all_witnesses(),
-                          multivariates.get_all_witnesses(),
-                          zk_sumcheck_data.value().masking_terms_evaluations)) {
-                bb::Univariate<FF, 2> edge({ multivariate[edge_idx], multivariate[edge_idx + 1] });
-                extended_edge = edge.template extend_to<MAX_PARTIAL_RELATION_LENGTH>();
-                extended_edge += masking_univariate;
-            };
-            // extend edges of public polynomials
-            for (auto [extended_edge, multivariate] :
-                 zip_view(extended_edges.get_non_witnesses(), multivariates.get_non_witnesses())) {
-                bb::Univariate<FF, 2> edge({ multivariate[edge_idx], multivariate[edge_idx + 1] });
-                extended_edge = edge.template extend_to<MAX_PARTIAL_RELATION_LENGTH>();
-            };
+    template <typename ProverPolynomialsOrPartiallyEvaluatedMultivariates>
+    void extend_edges_with_masking(ExtendedEdges& extended_edges,
```

Comment:
split the method into two to avoid optional access that was causing problems in wasm multi-threading